### PR TITLE
repaint when loaded out of viewport

### PR DESCRIPTION
**Pull request by collin**

### CODE REVIEW

**Comment by kand** [comment]
:+1:

**Comment by collin** [comment]
@kand now with tests :construction_worker:

File: dist/onion-sticky.browserify-shim.js [hunk 1]
```diff
@@ -95,7 +95,7 @@ var OnionSticky = (function () {
   }, {
     key: 'calculateScrollDirection',
     value: function calculateScrollDirection() {
-      if (!this._lastScrollTop) {
+      if (this._lastScrollTop === undefined) {
```
Comment (kand) [inline dist/onion-sticky.browserify-shim.js:98]: Use `typeof(this._lastScrollTop) === 'undefined'`, slightly more robust.

File: src/onion-sticky.js [hunk 2]
```diff
@@ -72,7 +72,7 @@ export class OnionSticky {
   }
 
   calculateScrollDirection () {
-    if (!this._lastScrollTop) {
+    if (this._lastScrollTop === undefined) {
```
Comment (kand) [inline src/onion-sticky.js]: Oops, commented about this in dist.

File: test/onion-sticky.spec.js [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+
+    it('false when element not fixed to bottom', function () {
+      this.sticky.getFrameData();
+
+      assert.isFalse(this.sticky.elementFixedToBottom());
+    });
+  });
+
+  describe('calculateScrollDirection', function () {
+    it('SCROLLING_UNCH on first call', function () {
```
Comment (kand) [inline test/onion-sticky.spec.js:208]: UNCH?

**Comment by kand** [comment]
Didn't notice before that this was in es6, nice! Also, I do like the makefile approach.

**Comment by kand** [comment]
:+1:

File: test/onion-sticky.spec.js (hunk 3 above)
Comment (collin) [inline test/onion-sticky.spec.js:208]: I have been infected by watching stock market reports as a child. changing to: `UNCHANGED`

**Pull request merged**